unshare/unshare.c

```diff
@@ -59,6 +83,120 @@ static void _check_proc_sys_file(const char *path)
 	}
 }
 
+static char **parse_proc_stringlist(const char *list) {
+	int fd, n, i, n_strings;
+	char *buf, **ret;
+	size_t size, new_size, used;
+
+	fd = open(list, O_RDONLY);
+	if (fd == -1) {
+		return NULL;
+	}
+	buf = NULL;
+	size = 0;
+	used = 0;
+	for (;;) {
+		new_size = used + BUFSTEP;
+		buf = realloc(buf, new_size);
```

Comment:
it will probably trigger a warning in the static analysis as we don't free the previous value of `buf` in case of failure

Reply:
Worth fixing, then.